src/widget/wtrackmenu.cpp

```diff
@@ -729,17 +728,40 @@ TrackIdList WTrackMenu::getTrackIds() const {
     return trackIds;
 }
 
-TrackPointerList WTrackMenu::getTrackPointers(
-        int maxSize) const {
+QList<TrackRef> WTrackMenu::getTrackRefs() const {
+    QList<TrackRef> trackRefs;
+    if (m_pTrackModel) {
+        trackRefs.reserve(m_trackIndexList.size());
+        for (const auto& index : m_trackIndexList) {
```

Comment:
m_trackIndexList is sharing due to WTrackMenu::getTrackIndices so we need a qAsConst() here. 
Can you fix that in the other loops as well? 
WTrackMenu::getTrackIndices can return a reference.

Reply:
Not needed. All methods are `const` and `m_trackIndexList` inherits this property.

Reply:
Ah, I didn't know that. Thank you for the education. :-)